### ISSUE TITLE
postfactor checks

### ISSUE THREAD

**Issue by twshimwell** [lofar-astron/factor#52]
Nice to check the flux levels (perhaps compare point like sources with TGSS) and positions of sources (can prob use TGSS again but perhaps FIRST nicer) for each facet and perhaps some other things too (maybe like source counts on the entire mosaic). I think Wendy already probably has some scripts to do this. 

Not sure if these checks should be incorporated into factor (i.e to correct the astrometry as going through) or run after factor (for comparison I think Huib has the astrometry check within spam). Good to catch errors as soon as possible though.

**Comment by mhardcastle** [comment]
I would suggest checking (and correcting) the flux scale before factor rather than afterwards. This can be done (and is actually easier) with low-resolution images and, if you want to get the frequency-dependent corrections even approximately right in the absence of a proper fix for the beam model problems, it needs to be done with fluxes from across the HBA band, not just the ~ 150-MHz ones. So the obvious thing to do would be to incorporate it pre-subtraction. Assuming that direction-dependent effects don't make that impossible.

**Comment by twshimwell** [comment]
Yeh this could be the way to do it. Presently our images in the subtract pipeline are not primary beam corrected but I guess thats a minorish alteration to change that (I think the new wsclean even has a lofar beam model). Perhaps a concern is that at the resolution of the high2 images I think that we start losing flux if the ionospheric conditions are just a little below average. Even in reasonable ionospheric conditions I suspect the high2 images systematically underestimate the peak flux of sources but the integrated flux is prob a little better. From the cycle 2 and 3 Tier-1 direction independent images I typically find that for about 75% of the fields the flux (from 120-168MHz) is within 20% of say TGSS at approximately the resolution of the high2 images (I can make some plots). I gather that cycle 4 and 5 are better and perhaps the number will be about 85%.

**Comment by mhardcastle** [comment]
Yeah, I'd do it with integrated flux to mitigate against that (but you are right about PB correction of course -- needs an additional step rather than just using the subtraction images).

**Comment by rvweeren** [comment]
Before doing work (writing code) on flux-checking I would wait until we have the "updated" HBA beam model available. (was that not supposed to be available around this time by now?)

**Comment by twshimwell** [comment]
Yeh im not sure how long we will wait for an improved beam model. Tammo presented on it at the LOFAR meeting last week and there are still a few issues (I don't think the talks are online yet unfortunately) which are not clear how to solve. Perhaps Tammo can comment further.

**Comment by tammojan** [comment]
There is an implementation of the beam model, just no mechanism to enable it through parsets etc. I could make a branch available with this model for testing. Shall I just do that?

**Comment by twshimwell** [comment]
OK, that wasn't very long to wait :)  Do you also have some notes or something on how reliable the new model will be in various different circumstances so that we know what to be careful about when using it.

**Comment by mhardcastle** [comment]
Yes it would make a big difference if we had a reliable beam model fix: in that case we could move from flux _correction_, as in my paper, to flux checking.
